src/Loop.php

```diff
@@ -33,6 +47,20 @@ public static function execute(callable $callback, LoopDriver $driver)
     }
 
     /**
+     * Create a new driver if a factory is present, otherwise throw.
+     *
+     * @throws \LogicException if no factory is set
+     */
+    private static function createDriver()
+    {
+        if (self::$factory === null) {
+            throw new \LogicException("Can't create an event loop driver without a factory.");
+        }
+
+        return self::$factory->create();
```

Comment:
This is where PHP 7 return types would be handy. 😄

I suppose would could emulate that behavior and check for a `LoopDriver` instance and throw if something else is returned.